libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/audioguidance/AudioGuidanceViewModel.kt

```diff
@@ -22,6 +24,7 @@ sealed class AudioAction {
  */
 @ExperimentalPreviewMapboxNavigationAPI
 class AudioGuidanceViewModel(
+    val navigationStateViewModel: NavigationStateViewModel,
```

Comment:
NIT: I would like us to be able to create components to maintain consistency, so that we don't inject viewmodels into viewmodels.

Reply:
Yeah I agree but the solutions I found need more exploring. 

As it is now, separating it into a component does not fix the dependency.
``` kotlin
val viewModel AudioGuidanceViewModel()
val component = AudioGuidanceComponent(
  AudioGuidanceViewModel,
  NavigationStateViewModel < === this is needed so we can disable RoutePreview 
)
```

Considering 1: Build a `ConditionalViewModel(conditional: Flow<Boolean>, UIViewModel)`. This could be a good idea, because DropInNavitationView can define conditions for ViewModels. It would be similar to what tomasz is introducing with `UIComponent `. But I'm considering it out of scope for now because there are many other issues being addressed. 

Considering 2: Use a component to disable `AudioGuidanceViewModel`. This will definitely not work with Android Auto because Android Auto works without any Views

Considering 3: We could add the RoutePreview state to MapboxNavigation. But that's definitely out of scope of this PR.

Reply:
```
val viewModel AudioGuidanceViewModel()
val component = AudioGuidanceComponent(
  AudioGuidanceViewModel,
  NavigationStateViewModel < === this is needed so we can disable RoutePreview 
)
```
This is fine correct as a component can have as many viewmodels as dependencies. If we do this, then the `onAttached` inside `AudioGuidanceViewModel` will move to `AudioGuidanceComponent`